src/CMakeLists.txt

```diff
@@ -1,78 +1,12 @@
-set_source_files_properties(${PROJECT_SOURCE_DIR}/third_party/jsoncpp/jsoncpp.cpp PROPERTIES COMPILE_FLAGS -w)
-set_source_files_properties(${PROJECT_SOURCE_DIR}/src/p11engine.cc PROPERTIES COMPILE_FLAGS -Wno-deprecated-declarations)
-
-add_subdirectory("utilities")
-add_subdirectory("asn1")
-add_subdirectory("uptane")
-add_subdirectory("socket_activation")
-add_subdirectory("secondary_ipc")
-add_subdirectory("package_manager")
-add_subdirectory("telemetry")
-
-# set main aktualizr source files
-set(AKTUALIZR_SOURCES aktualizr.cc
-            bootstrap.cc
-            commands.cc
-            config.cc
-            events.cc
-            eventsinterpreter.cc
-            gatewaymanager.cc
-            ipsecondarydiscovery.cc
-            socketgateway.cc
-            sotauptaneclient.cc
-            timer.cc
-            )
-
-# set headers used for clang format
-set(HEADERS aktualizr.h
-            bootstrap.h
-            channel.h
-            commands.h
-            config.h
-            events.h
-            eventsinterpreter.h
-            exceptions.h
-            fsstorage.h
-            gateway.h
-            gatewaymanager.h
-            httpinterface.h
-            invstorage.h
-            ipsecondarydiscovery.h
-            logging.h
-            openssl_compat.h
-            socketgateway.h
-            sotauptaneclient.h
-            sqlstorage.h
-            sql_utils.h
-            timer.h)
-
-set(STORAGE_SOURCES fsstorage.cc sqlstorage.cc invstorage.cc)
-add_library(storage OBJECT ${STORAGE_SOURCES})
-
-add_library(jsoncpp OBJECT ${PROJECT_SOURCE_DIR}/third_party/jsoncpp/jsoncpp.cpp)
-
-add_library(aktualizr_static_lib STATIC
-    ${AKTUALIZR_SOURCES}
-    $<TARGET_OBJECTS:asn1>
-    $<TARGET_OBJECTS:jsoncpp>
-    $<TARGET_OBJECTS:package_manager>
-    $<TARGET_OBJECTS:utilities>
-    $<TARGET_OBJECTS:aktualizr_secondary_ipc>
-    $<TARGET_OBJECTS:storage>
-    $<TARGET_OBJECTS:uptane>
-    $<TARGET_OBJECTS:socket_activation>
-    $<TARGET_OBJECTS:telemetry>)
-
-target_include_directories(aktualizr_static_lib PUBLIC
-    $<TARGET_PROPERTY:package_manager,INCLUDE_DIRECTORIES>)
+add_subdirectory("libaktualizr")
 
 set(OPCUABRIDGE_SOURCES opcuabridge/common.cc
                         opcuabridge/filelist.cc
                         opcuabridge/opcuabridgeclient.cc
                         opcuabridge/opcuabridgeserver.cc
                         opcuabridge/opcuabridgediscoveryclient.cc
                         opcuabridge/opcuabridgediscoveryserver.cc
-                        uptane/opcuasecondary.cc)
+                        libaktualizr/uptane/opcuasecondary.cc)
```

Comment:
I think opcuabridge should link with libaktualizr and dont share sources.

Reply:
Fixed